### DISCUSSION TITLE
Ethernet

### DISCUSSION THREAD

**Discussion by danielswe88** [taligentx/dscKeybusInterface#58] [dscKeybusInterface Discussion]
Hi!

I have an ethernet module connected to my DSC alarm already, is there any possibility to connect to that one to get the integration to my homey? It feels a little scary to fiddle around with the cables on the alarm PCB.

I am currently using the "PowerSeries Neo Go" app, but it is not working good at all, and I want to integreate my DSC alarm with the rest of my home automation.

Thanks in advance!
/Daniel

**Comment by taligentx** [comment]
Hi @danielswe88 - DSC Neo series uses a different data protocol that is encrypted, so it is not supported by this interface.  Unfortunately the only option is to use the DSC provided accessories.  Thanks for taking a look!

**Comment by danielswe88** [comment]
Hello @taligentx.

I did a lot of research about this and found out exactly what you are saying. I really appreciate your work on this project. Unfortunately I can't get that to work due to my hardware.

I was talking to my electrician (it's a newly built house), he said he could come and remove our PowerSeries NEO and instead install an older DSC with nothing in exchange. But I don't know... It feels so and so going back to older hardware, maybe I could wait until DSC launches some kind of API for the NEO series. Or I could try to reverse engineer the API...

/ Daniel

**Comment by taligentx** [comment]
I've only taken a quick look but it seems that there are only a few options like [Control4](https://www.control4.com/solutions/whole-home) for integration with the [DSC TL280 Internet Communicator module](https://www.dsc.com/alarm-security-products/TL280%20-%20Internet%20Alarm%20Communicator/2487).

The TL280R version has an RS232 serial interface so it could be possible to reverse engineer the serial API, though I haven't seen any documentation on it.

The Neo series seems to be what DSC wants to migrate to completely, but the PowerSeries has such a large installed base that it's hard to see third parties dropping support anytime soon.  One advantage of the PowerSeries is that you would have a wider choice of third party interfaces - Envisalink, Alarm Decoder, etc.  But it's also a matter of home security so carefully considering the pros/cons is absolutely warranted.

**Comment by fariddergal** [comment]
Hi 
Any news about Neo interface?

**Comment by Dilbert66** [comment]
No.  If you have a neo, you can contribute  with logic analyzer traces of the bus.  None of the developers here has a neo system to anazlyze so it's up to someone with physical access to assist and provide traces/and or decode the bus before any coding can occur.

**Reply by fariddergal** [reply]
The neo system uses a communication board tl280 which connects the system to the app connect alarm , this board has also serial communication I think I could read this traces but I have no idea about how to translate it

**Reply by danielswe88** [reply]
Good morning!

I have a Neo. My main board is a HS2064 and my network card is a TL280E. I am working as a full stack developer and. I think I would like to connect a serial bus to my alarm. So you say it is possible to do against my network card and then just connect that serial cable into a raspberry pi? I think that is how I would like to hook my home alarm up. When I have done so I guess I could help you figuring out the codes?

**Reply by fariddergal** [reply]
Hi Daniel 

That sound good 
Any way to link the alarm to our HA is good 
Thanks

**Comment by fariddergal** [comment]
> Hi!
> 
> I have an ethernet module connected to my DSC alarm already, is there any possibility to connect to that one to get the integration to my homey? It feels a little scary to fiddle around with the cables on the alarm PCB.
> 
> I am currently using the "PowerSeries Neo Go" app, but it is not working good at all, and I want to integreate my DSC alarm with the rest of my home automation.
> 
> Thanks in advance!
> /Daniel

Hi Daniel 

I  have the tl280e with the app connect alarm and in the app all the information of the alarm is available as status if the sensors , activity in general etc 
I was wondering if you know if is possible to get this data from the app to use it for the home assistant?

**Reply by danielswe88** [reply]
Hi! I didn't fully read what you said there last time.

As I understand you are using the "new app" which is called "Connect Alarm". To be able to use this app it seems like I have to re-configure my panel so that it uses another base url. As I understand it this setting (base url) is putted in two places in the panel: one for real time events and one place for something else.

What if we there, instead, putted a proxy, a small web service which forwards all requests back and forth to tyco? Then we could learn enough about the system so that we could maybe re-use that functionality ourselves? Just a question :) 

I have at least sent my installer a mail so that I can get the installer code so that I can "fix" my panel so that it uses the new api so I can use the new app.

**Reply by fariddergal** [reply]
Hi Daniel 

Sound good option 
The thing here is that all this information send from the comunication  board to the Tyco server can be re used for our HA 
Also this information is used by monitoring companies so it has to 3xist a way to understand and interpretaste the information
Unfortunately I am a HW engineer and I don't know how it can be done by SW

**Reply by danielswe88** [reply]
I am working as a software engineer.

How I understand how the alarm is working is that you can configure it to have two different servers to send data to. The first one would be the tyco monitor thing, which is used for the new app? And the second one could be a security company which I could pay so that they can monitor my house (fire alarm and security alarm)?

Since I don't have number two in above I guess I then could point the server two against my own server. If I then, on my own server, build something like an API which catches everything on every port and logs it into a database, I would then hopefully get an understanding of how the alarm communicates. The next step here would be that I configure this server I have at home to have an API that homebridge could communicate with.

Because of full time work and kids I have no idea if and how I will find time for this though, but how does the methods sound to you?

**Reply by fariddergal** [reply]
Hi Daniel 

I have made so e research and there is a way to use the second communicator to send messages to the monitoring companies, by having the IP address of a server and making the correct configuration in the panel, is possible to see the messages , there is a web from DSC to make test using their server to make sure that everything is working properly, now if we could emulate a server to receive these messages I don't know is could work to then send them to HA server

**Reply by danielswe88** [reply]
That's nice!

I have still not have had time to look at this, but since golang is my preferred language of choice, I did a quick google and found this proxy. It looks to me like it is forwarding all request with same headers, payload etc, but it is logging each request:

https://gist.github.com/yowu/f7dc34bd4736a65ff28d

So if we would manage to run this locally at home (`go run main.go`) and then look at the logs of this web server while the home alarm is making requests to it, we would maybe better understand the different calls /  endpoints.

**Comment by fariddergal** [comment]
Hi

Yes sounds good to use the second port to use it for the home A, lets see
if that works.
